[URGENT] error: no member named 'for_each_n' in namespace 'std'

https://github.com/ROCm/MIOpen/blob/9122863720a32caefa9a2303b61a6e0b36fcae3d/src/graphapi/opgraph.cpp#L352

On RHEL and SLES distributions, we are getting:
```
[2024-05-17T11:27:20.969Z] /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/MLOpen/src/graphapi/opgraph.cpp:352:18: error: no member named 'for_each_n' in namespace 'std'

[2024-05-17T11:27:20.969Z]   352 |             std::for_each_n(static_cast<miopenBackendDescriptor_t*>(arrayOfElements),
```

We need to resolve this because staging is blocked due to the issue

@atamazov two issues I have observed:
```
[2024-05-17T11:27:20.969Z] /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/MLOpen/src/graphapi/opgraph.cpp:352:18: error: no member named 'for_each_n' in namespace 'std'
[2024-05-17T11:27:20.969Z]   352 |             std::for_each_n(static_cast<miopenBackendDescriptor_t*>(arrayOfElements),
[2024-05-17T11:27:20.969Z]       |             ~~~~~^
[2024-05-17T11:27:20.969Z] /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/MLOpen/src/graphapi/opgraph.cpp:356:36: error: variable 'backendDescriptor' cannot be implicitly captured in a lambda with no capture-default specified
[2024-05-17T11:27:20.969Z]   356 |                                 if(backendDescriptor.isFinalized())
[2024-05-17T11:27:20.969Z]       |                                    ^
```

--- a/requirements.txt
+++ b/requirements.txt
@@ -7,4 +7,4 @@ nlohmann/json@v3.11.2 -DJSON_MultipleHeaders=ON -DJSON_BuildTests=Off
 ROCm/FunctionalPlus@v0.2.18-p0
 ROCm/eigen@3.4.0
 ROCm/frugally-deep@9683d557eb672ee2304f80f6682c51242d748a50
-ROCm/composable_kernel@9b3c4ac4754c7e3d6847e00cb3b553f4e98a747d -DCMAKE_BUILD_TYPE=Release -DINSTANCES_ONLY=ON
+ROCm/composable_kernel@a6ef5c391e893c423a34c7a80d99009ddf762af4 -DCMAKE_BUILD_TYPE=Release -DINSTANCES_ONLY=ON

--- a/src/graphapi/opgraph.cpp
+++ b/src/graphapi/opgraph.cpp
@@ -349,20 +349,20 @@ void BackendOperationGraphDescriptor::setAttribute(miopenBackendAttributeName_t 
             std::vector<OpNode*> nodes;
             nodes.reserve(elementCount);
 
-            std::for_each_n(static_cast<miopenBackendDescriptor_t*>(arrayOfElements),
-                            elementCount,
-                            [&descriptors, &nodes](miopenBackendDescriptor_t apiDescriptor) {
-                                BackendDescriptor& backendDescriptor = deref(apiDescriptor);
-                                if(backendDescriptor.isFinalized())
-                                {
-                                    descriptors.push_back(apiDescriptor);
-                                    nodes.push_back(backendDescriptor.getOperation());
-                                }
-                                else
-                                {
-                                    MIOPEN_THROW(miopenStatusBadParm);
-                                }
-                            });
+            std::for_each(static_cast<miopenBackendDescriptor_t*>(arrayOfElements),
+                          static_cast<miopenBackendDescriptor_t*>(arrayOfElements) + elementCount,
+                          [&descriptors, &nodes](miopenBackendDescriptor_t apiDescriptor) {
+                              BackendDescriptor& backendDescriptor = deref(apiDescriptor);
+                              if(backendDescriptor.isFinalized())
+                              {
+                                  descriptors.push_back(apiDescriptor);
+                                  nodes.push_back(backendDescriptor.getOperation());
+                              }
+                              else
+                              {
+                                  MIOPEN_THROW(miopenStatusBadParm);
+                              }
+                          });
 
             if(!internal::noRepetitions(nodes))
             {

